src/java.desktop/share/classes/javax/swing/JSplitPane.java

```diff
@@ -361,6 +362,15 @@ public JSplitPane(int newOrientation,
 
     }
 
+    /**
+     * {@inheritDoc}
+     * @param enabled {@inheritDoc}
+     */
+    @Override
+    public void setEnabled(boolean enabled) {
+        super.setEnabled(enabled);
+        ((BasicSplitPaneUI)(this.getUI())).getDivider().setEnabled(enabled);
```

Comment:
is it fine to assume that the UI is a BasicSplitPaneUI object? what happens if the JSplitPane UI is an AquaSplitPaneUI for example?

Reply:
All L&F including Aqua extends BasicSplitPaneUI so it's ok...The existing test iterates through all L&F without any issue..

Reply:
> All L&F including Aqua extends BasicSplitPaneUI so it's ok...The existing test iterates through all L&F without any issue..

That is true, yet it is still possible to set a L&F that doesn't extend `BasicSplitPaneUI` and the updated code will throw `ClassCastException`.

I'm sure such a situation is rare, if it exists at all, yet I don't think the public API should have such a limitation: `JSplitPane.setUI` accepts `SplitPaneUI`, so it is valid to pass an object that is not subclass of `BasicSplitPaneUI`.

Reply:
ok...Will add check..